src/mpsc/blocking.rs

```diff
@@ -351,6 +351,124 @@ feature! {
             }
         }
 
+        /// Reserves a slot in the channel to mutate in place, blocking until
+        /// there is a free slot to write to, waiting for at most `timeout`.
+        ///
+        /// This is similar to the [`send`] method, but, rather than taking a
+        /// message by value to write to the channel, this method reserves a
+        /// writable slot in the channel, and returns a [`SendRef`] that allows
+        /// mutating the slot in place. If the [`StaticReceiver`] end of the
+        /// channel uses the [`StaticReceiver::recv_ref`] method for receiving
+        /// from the channel, this allows allocations for channel messages to be
+        /// reused in place.
+        ///
+        /// # Errors
+        ///
+        /// - [`Err`]`(`[`SendTimeoutError::Timeout`]`)` if the timeout has elapsed.
+        /// - [`Err`]`(`[`SendTimeoutError::Closed`]`)` if the channel has closed.
+        ///
+        /// # Examples
+        ///
+        /// Sending formatted strings by writing them directly to channel slots,
+        /// in place:
+        ///
+        /// ```
+        /// use thingbuf::mpsc::{blocking::StaticChannel, errors::SendTimeoutError};
+        /// use std::{fmt::Write, time::Duration, thread};
+        ///
+        /// static CHANNEL: StaticChannel<String, 1> = StaticChannel::new();
+        /// let (tx, rx) = CHANNEL.split();
+        ///
+        /// thread::spawn(move || {
+        ///     thread::sleep(Duration::from_millis(500));
+        ///     let msg = rx.recv_ref().unwrap();
+        ///     println!("{}", msg);
+        ///     thread::sleep(Duration::from_millis(500));
+        /// });
+        ///
+        /// thread::spawn(move || {
+        ///     let mut value = tx.send_ref_timeout(Duration::from_millis(200)).unwrap();
+        ///     write!(value, "hello").expect("writing to a `String` should never fail");
+        ///     thread::sleep(Duration::from_millis(400));
+        ///
+        ///     let mut value = tx.send_ref_timeout(Duration::from_millis(200)).unwrap();
+        ///     write!(value, "world").expect("writing to a `String` should never fail");
+        ///     thread::sleep(Duration::from_millis(400));
+        ///
+        ///     assert_eq!(
+        ///         Err(&SendTimeoutError::Timeout(())),
+        ///         tx.send_ref_timeout(Duration::from_millis(200)).as_deref().map(String::as_str)
+        ///     );
+        /// });
+        /// ```
+        ///
+        /// [`send`]: Self::send
```

Comment:
and, if we address my above suggestion, this should probably be
```suggestion
        /// [`send_timeout`]: Self::send_timeout
```